modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -49,6 +50,24 @@
  * This directory holds files for checkpoint start and end.
  */
 public class CheckpointHistory {
+    /** The message appears when no one checkpoint was not reserved for cache. */
+    private static final String NOT_RESERVED_WAL_REASON = "If checkpoint WAL history can't be reserved";
+
+    /** The message puts down to log when an exception happened during reading reserved WAL. */
+    private static final String WAL_SEG_CORRUPTED_REASON = "Segment corrupted";
+
+    /** Reason means no more history reserved for the cache. */
+    private static final String NO_MORE_HISTORY_REASON = "No more history for reservation";
```

Comment:
Reserved checkpoint is first in history

Reply:
That reason means we have not any checkpoints in WAL history.
Ma be "History exhausted"

Reply:
Reserved checkpoint is the oldest in history.